lxd/devlxd.go

```diff
@@ -173,18 +173,51 @@ var devlxdEventsGet = devLxdHandler{"/1.0/events", func(d *Daemon, c instance.In
 	return resp
 }}
 
-var devlxdAPIGet = devLxdHandler{"/1.0", func(d *Daemon, c instance.Instance, w http.ResponseWriter, r *http.Request) response.Response {
-	location := "none"
-	clustered, err := cluster.Enabled(d.db.Node)
-	if err != nil {
-		return response.DevLxdErrorResponse(api.StatusErrorf(http.StatusInternalServerError, "internal server error"), c.Type() == instancetype.VM)
-	}
+var devlxdAPIHandler = devLxdHandler{"/1.0", func(d *Daemon, c instance.Instance, w http.ResponseWriter, r *http.Request) response.Response {
+	if r.Method == "GET" {
+		location := "none"
+		clustered, err := cluster.Enabled(d.db.Node)
+		if err != nil {
+			return response.DevLxdErrorResponse(api.StatusErrorf(http.StatusInternalServerError, "internal server error"), c.Type() == instancetype.VM)
+		}
+
+		if clustered {
+			location = c.Location()
+		}
+
+		var state string
+
+		if shared.IsTrue(c.LocalConfig()["volatile.last_state.ready"]) {
+			state = "READY"
+		} else {
+			state = "STARTED"
+		}
 
-	if clustered {
-		location = c.Location()
+		return response.DevLxdResponse(http.StatusOK, devlxdGet{APIVersion: version.APIVersion, Location: location, InstanceType: c.Type().String(), devlxdPut: devlxdPut{State: state}}, "json", c.Type() == instancetype.VM)
+	} else if r.Method == "PATCH" {
+		req := devlxdPut{}
+
+		err := json.NewDecoder(r.Body).Decode(&req)
+		if err != nil {
+			return response.DevLxdErrorResponse(api.StatusErrorf(http.StatusBadRequest, err.Error()), c.Type() == instancetype.VM)
+		}
+
+		state := strings.ToUpper(req.State)
+
+		if state != "STARTED" && state != "READY" {
+			return response.DevLxdErrorResponse(api.StatusErrorf(http.StatusForbidden, "Invalid state %q", state), c.Type() == instancetype.VM)
```

Comment:
I think `http.StatusBadRequest` would be more appropriate, as its failed validation effectively, rather than that particular user not having the permissions to do the update.